tests/test_slim_engine.py

```diff
@@ -3079,6 +3079,100 @@ def test_errors(self):
             stdpopsim.selection_coeff_from_mutation(ts, "bar")
 
 
+@pytest.mark.skipif(IS_WINDOWS, reason="SLiM not available on windows")
+class TestPopSizes:
+    """
+    Test that population sizes are set correctly.
+    Thanks to discretization artifacts, this is a bit tricky in phases
+    of exponential growth: suppose we are growing at rate r for the
+    time period [t, t+T), starting from size x. In discrete time, at
+    time t we set the size to x; then subsequently we re-set the size
+    only T-1 times. (One consideration is that we re-set sizes in `late`,
+    which doesn't take effect until the next tick, so a reset in the final
+    time step has no effect.) This means that the continuous-time msprime model
+    that we base this on is growing for total time T, but the SLiM model grows
+    in T-1 time steps. Furthermore, we'd like both the final and initial sizes
+    to be exactly what they are supposed to be. This amounts to approximating
+    exp(r*t) over the interval [0,T] by exp(r*t*T/(T-1)). Probably something
+    nicer could be done, but this is close. The approximation is better for
+    smaller growth rates.
+    """
+
+    def verify_pop_sizes(self, model, samples, generation_time=1, Q=1):
+        species = stdpopsim.get_species("HomSap")
+        contig = species.get_contig("chr17", left=1e7, right=1e7 + 1e4)
+        engine = stdpopsim.get_engine("slim")
+        sp_model = stdpopsim.DemographicModel(
+            id="test",
+            description="foo",
+            long_description="fooooooo",
+            generation_time=generation_time,
+            model=model,
+        )
+        ts = engine.simulate(
+            sp_model,
+            contig,
+            samples,
+            seed=123,
+            slim_burn_in=0.0,
+            verbosity=3,
+            slim_scaling_factor=Q,
+        )
+        slim_sizes = ts.metadata["SLiM"]["user_metadata"]["population_sizes"][0]
+        dd = model.debug()
+        T = ts.metadata["SLiM"]["cycle"]
+        for pop in slim_sizes:
+            k = [x.name for x in model.populations].index(pop)
+            pop_t = np.array(slim_sizes[pop][0]["t"])
+            pop_N = np.array(slim_sizes[pop][0]["N"])
+            dd_N = dd.population_size_trajectory(np.floor((T - pop_t) * Q))[:, k] / Q
+            dd_N1 = (
+                dd.population_size_trajectory(np.floor((T - pop_t + 1) * Q))[:, k] / Q
+            )
+            # np.testing.assert_allclose(pop_N, dd_N, rtol=0.01)
+            assert np.all(
```

Comment:
Worth leaving a note saying the double check here is to deal with discretization error